packages/patternfly-3/patternfly-react/src/components/DateTimePicker/DateComponents/DateInput.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { addMonths } from './helpers';
+import MonthView from './MonthView';
+import YearView from './YearView';
+import DecadeView from './DecadeView';
+
+class DateInput extends React.Component {
+  state = {
+    selectedDate: new Date(this.props.date),
+    date: new Date(this.props.date),
+    typeOfDateInput: this.props.typeOfDateInput
+  };
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const nextDate = new Date(nextProps.date);
+    const prevDate = new Date(prevState.selectedDate);
+    const nextType = new Date(nextProps.date);
+    return Date.parse(nextDate) === Date.parse(prevDate)
+      ? null
+      : {
+          selectedDate: nextDate,
+          date: nextDate,
+          typeOfDateInput: nextType
+        };
+  }
+  getPrevMonth = () => {
+    const { date } = this.state;
+    this.setState({ date: addMonths(date, -1) });
+  };
+  getNextMonth = () => {
+    const { date } = this.state;
+    this.setState({ date: addMonths(date, 1) });
+  };
+  setSelected = day => {
+    this.setState({
+      selectedDate: day,
+      date: day
+    });
+    this.props.setSelected(day);
+  };
+  toggleDateView = (type = null) => {
+    this.setState({
+      typeOfDateInput: type
+    });
+  };
+  render() {
+    const { date, locale, weekStartsOn, className, setSelected } = this.props;
+    const { typeOfDateInput } = this.state;
+    return (
+      <div className={`datepicker ${className}`}>
+        {(() => {
+          switch (typeOfDateInput) {
```

Comment:
could be moved to an helper file into a function like: `getDateViewByType` ?

Reply:
I moved it into a function. I think it should stay in the DateInput file and shouldn't be in a helper file because only the DateInput component will use it.